src/ImageSharp/Common/Helpers/SimdUtils.Shuffle.cs

```diff
@@ -0,0 +1,271 @@
+        /// <param name="dest">The destination span of floats.</param>
+        /// <param name="control">The byte control.</param>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static void Shuffle4Channel(
+            ReadOnlySpan<float> source,
+            Span<float> dest,
+            byte control)
+        {
+            VerifyShuffleSpanInput(source, dest);
+
+            // TODO: There doesn't seem to be any APIs for
+            // System.Numerics that allow shuffling.
+#if SUPPORTS_RUNTIME_INTRINSICS
+            HwIntrinsics.Shuffle4ChannelReduce(ref source, ref dest, control);
+#endif
+
+            // Deal with the remainder:
+            if (source.Length > 0)
+            {
+                ShuffleRemainder4Channel(source, dest, control);
+            }
+        }
+
+        /// <summary>
+        /// Shuffle 8-bit integers in a within 128-bit lanes in <paramref name="source"/>
+        /// using the control and store the results in <paramref name="dest"/>.
+        /// </summary>
+        /// <param name="source">The source span of bytes.</param>
+        /// <param name="dest">The destination span of bytes.</param>
+        /// <param name="control">The byte control.</param>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static void Shuffle4Channel(
+            ReadOnlySpan<byte> source,
+            Span<byte> dest,
+            byte control)
+        {
+            VerifyShuffleSpanInput(source, dest);
+
+            // TODO: There doesn't seem to be any APIs for
+            // System.Numerics that allow shuffling.
+#if SUPPORTS_RUNTIME_INTRINSICS
+            HwIntrinsics.Shuffle4ChannelReduce(ref source, ref dest, control);
+#endif
+
+            // Deal with the remainder:
+            if (source.Length > 0)
+            {
+                ShuffleRemainder4Channel(source, dest, control);
+            }
+        }
+
+        [MethodImpl(InliningOptions.ColdPath)]
+        public static void ShuffleRemainder4Channel(
+            ReadOnlySpan<float> source,
+            Span<float> dest,
+            byte control)
+        {
+            ref float sBase = ref MemoryMarshal.GetReference(source);
+            ref float dBase = ref MemoryMarshal.GetReference(dest);
+            Shuffle.InverseMmShuffle(control, out int p3, out int p2, out int p1, out int p0);
+
+            for (int i = 0; i < source.Length; i += 4)
+            {
+                Unsafe.Add(ref dBase, i) = Unsafe.Add(ref sBase, p0 + i);
+                Unsafe.Add(ref dBase, i + 1) = Unsafe.Add(ref sBase, p1 + i);
+                Unsafe.Add(ref dBase, i + 2) = Unsafe.Add(ref sBase, p2 + i);
+                Unsafe.Add(ref dBase, i + 3) = Unsafe.Add(ref sBase, p3 + i);
+            }
+        }
+
+        [MethodImpl(InliningOptions.ColdPath)]
+        public static void ShuffleRemainder4Channel(
+            ReadOnlySpan<byte> source,
+            Span<byte> dest,
+            byte control)
+        {
+#if NETCOREAPP
```

Comment:
The referenced PR is quite old. Can't confirm for 100% percent, but I think the change is also present in 64bit RyuJIT for .NET Framework. Without the ROTL / ROTL optimization the code should perform equivalently (or even slightly faster) to the `byte` indexing below.

I think we should remove this condition.

Reply:
Ok, will do. 👍